### ISSUE TITLE
☂️ Improvements to Vertical question step

### ISSUE THREAD

**Issue by taipeicoder** [Automattic/wp-calypso#62582]
## Description

Here we list the proposed improvements that can be done to the Vertical question step (`/stepper/vertical`).

### From https://github.com/Automattic/wp-calypso/pull/62365#issuecomment-1091118044:

- [x] Add a "Skip to My Home" button in the intent screen.
<img width="1083" alt="Screen Shot 2022-04-07 at 4 08 54 pm" src="https://user-images.githubusercontent.com/22446385/162131660-c757da74-fcc8-4f82-b72f-41ec613e7bc8.png">

- [x] ~~The "Something Else" option is hidden by a scrollbar.~~
<img width="545" alt="Screen Shot 2022-04-07 at 4 10 26 pm" src="https://user-images.githubusercontent.com/22446385/162131927-5f0f78ec-15f8-44e2-97e5-607ede3ade4b.png">

- [x] Add a loading state to typing.

https://user-images.githubusercontent.com/22446385/162129927-85ac22ad-6608-40a2-b4d0-6e10329e2943.mov

- [x] Many of the suggestions don't seem tailored to website verticals (e.g. when I search for Design It would be good to prioritize common verticals such as "Web Design"). I think that there was some data available(?) around which verticals were the most popular, I wonder if we can make that available in the site-verticals API for prioritizing search results.

**Comment by arthur791004** [comment]
I found that the new stepper framework seems to have to send an API request to save the selected value directly. However, it might be a little weird if the user doesn't complete the flow. For example, if the user picks the vertical, go to the next step a, back to the vertical step, and then close the page. IMO, we have to keep the site vertical null as they don't finish the site setup flow. Does it make sense? Or at least, we have to show the loading status on the “Continue” button because I found it a little delayed when I clicked on the “Continue” button.
